linkerd/config/src/main/scala/io/buoyant/linkerd/config/Parser.scala

```diff
@@ -41,6 +54,19 @@ object Parser {
     val mapper = new ObjectMapper(factory) with ScalaObjectMapper
     mapper.registerModule(DefaultScalaModule)
     mapper.registerModule(customTypes)
+    mapper.setSerializationInclusion(Include.NON_NULL)
+
+    // Subtypes must not conflict
+    configInitializers.groupBy(_.configId).collect {
+      case (id, cis) if cis.size > 1 =>
+        throw ConflictingSubtypes(cis(0).namedType, cis(1).namedType)
+    }
+    for (ci <- configInitializers) ci.registerSubtypes(mapper)
+
     mapper
   }
+
+  def jsonObjectMapper(
+    configInitializers: Seq[ConfigInitializer]
+  ): ObjectMapper with ScalaObjectMapper = objectMapper("{", configInitializers)
```

Comment:
this is clever, but I think it would be more readable to add another method that takes a `JsonFactory` or `YamlFactory` and have both this and `objectMapper` call into that.